Fix non-json outputs

Coriou opened the pull request:
Sometimes youtube-dl won't return JSON. It can return a simple string (when using the "-g" or "--get-url" flags for instance). This commit, with the minimum amount of modifications possible, tries to handle this case.

Comes with a test

I think it's be cool to know what other flags can potentially output non-json results to test them all.

Discussed in #198

Kikobeats commented on lib/youtube-dl.js:
```diff
@@ -238,6 +238,10 @@ ytdl.exec = function exec(url, args, options, callback) {
  */
 function parseInfo(data) {
+  // Youtube-dl might return just an url as a string when using the "-g" or "--get-url" flag
+  if (typeof data === "string" && data.startsWith("http"))
```
not a requisite but appreciated if you can extract this into a function

```js
const isString = str => typeof str === "string"
```

and then use the function there and the rest of the places, like [here](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L191).

Coriou replied:
Done in e3f4a9d

Kikobeats commented on lib/youtube-dl.js:
```diff
@@ -238,6 +238,10 @@ ytdl.exec = function exec(url, args, options, callback) {
  * @returns {Object}
  */
 function parseInfo(data) {
+  // Youtube-dl might return just an url as a string when using the "-g" or "--get-url" flag
```
Little typo: `Youtube-dl` → `youtube-dl`

Coriou replied:
Done in e968072

Kikobeats commented on lib/youtube-dl.js:
```diff
@@ -316,6 +320,12 @@ ytdl.getInfo = function getInfo(url, args, options, callback) {
 
+    // If using the "-g" or "--get-url" flag youtube-dl will return just a string (the URL to the video) which messes up the parsing
+    // This fixes this behaviour
+    if (args && (args.indexOf("-g") > -1 || args.indexOf("--get-url") > -1))
```
The same thing here, this kind of checks are around the code, could be nice start extracting it to make the code more readable:

```js
const has = (collection, value) => collection && collectionindexOf(value)  > -1
```

Coriou replied:
Done in 5ec1f87

Left [this](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L313) and [this](https://github.com/przemyslawpluta/node-youtube-dl/blob/113ed2a1d61e434f35d7fa173770043c80704ddd/lib/youtube-dl.js#L376) untouched as they're checking for a special index didn't want to break anything that's beyond the scope of this PR already.

Coriou commented:
> LGTM; The only action to perform in order to merge this removes `package-lock.json` since we are not interested in maintaining it.

Removed in 56cbc63, my bad

> I added some comments most as recommendations, feel free to address them, very appreciated if you do that.

Addressed each of them in different commits, feel free to hand pick them

Coriou commented:
Not sure what's causing the last build to change, seems like Youtube isn't returning the expected format. Passing in Node 8-11 on my local machine

Kikobeats commented:
tests are not deterministic; 

That's a legacy thing and sometimes the video format can be different (webp, mp4).

Kikobeats commented:
Green! 🎉

Coriou commented:
Thank you for looking into this. Also fixes #198
